core/payment-driver/zksync/src/zksync/wallet.rs

```diff
@@ -184,3 +185,53 @@ async fn unlock_wallet<S: EthereumSigner + Clone>(wallet: Wallet<S>) -> Result<(
     }
     Ok(())
 }
+
+pub async fn withdraw<S: EthereumSigner + Clone>(
+    wallet: Wallet<S>,
+    amount: Option<BigDecimal>,
+) -> Result<SyncTransactionHandle, GenericError> {
+    let balance = wallet
+        .get_balance(BlockStatus::Committed, ZKSYNC_TOKEN_NAME)
+        .await
+        .map_err(GenericError::new)?;
+    info!(
+        "Wallet funded with {} tGLM available for withdrawal",
+        utils::big_uint_to_big_dec(balance.clone())
+    );
+
+    info!("Obtaining withdrawal fee");
+    let address = wallet.address();
+    let withdraw_fee = wallet
+        .provider
+        .get_tx_fee(TxFeeTypes::Withdraw, address, ZKSYNC_TOKEN_NAME)
+        .await
+        .map_err(GenericError::new)?
+        .total_fee;
+    info!(
+        "Withdrawal transaction fee {:.5}",
+        utils::big_uint_to_big_dec(withdraw_fee.clone())
+    );
+
+    let amount = amount
+        .ok_or(GenericError::new("unused"))
```

Comment:
Other options?
This is unused error just to satisfy type-check, I cannot use `Option::and_then` directly because fn used needs to return `Option` not `Result<>` as `utils::big_dec_to_big_uint` do.

So I map `Option` to `Result` - what kind of error is more appropriate here?
It won't be used anyway because of `.unwrap_or(balance.clone())` ...

WDYT :question:

Reply:
How about something along these lines?
```rust
let amount = match amount {
    Some(amount) => utils::big_dec_to_big_uint(amount)?,
    None => balance.clone(),
};
```